oeps/oep-0019-bp-developer-documentation.rst

```diff
@@ -0,0 +1,298 @@
++-----------------+--------------------------------------------------------+
+| `Review Period` | <start - target end dates for review>                  |
++-----------------+--------------------------------------------------------+
+| `Resolution`    | <links to any discussions where the final              |
+|                 | status was decided>                                    |
++-----------------+--------------------------------------------------------+
+
+.. contents::
+   :local:
+   :depth: 2
+
+*****************
+IN PROGRESS NOTES
+*****************
+
+The following section is a set of assorted notes for developing this OEP. This entire section should no longer exist at
+some point.  Definitely before accepting, and also potentially before full review.
+
+============================
+General Questions / Comments
+============================
+
+* This OEP might want to be broken down into more than one rST file.  One section of the OEP could include future ideas
+  for the OEP, since we don't know everything, but we want to land what we do know.
+
+   * Note that everything beyond the "IN PROGRESS NOTES" section is still fair game for adjustments. We may not want to
+     follow the full initial OEP format for this particular OEP.
+
+* Nimisha has notes on developer documentation in `this slide in the Architecture Onboarding Presentation`_.
+
+    * [Robert] EdX Courses for learning is probably beyond the scope of this first OEP.
+
+* Nate/Nimisha tested exchanging draw.io diagrams via XML and VSDX which are sharable via github.  However, we are
+  still using LucidChart. This is an interesting idea that shouldn't get lost, but premature for commitment in this
+  OEP.
+
+* From Nate:
+
+   * Idea: use `Algolia`_ to search ALL of these resources including edx-code, openedx-operations  Slack archive
+     and Confluence. Algolia is `free for open source projects`_.
+
+      * [Robert] We have no story for documentation discovery and I think this is wildly important, so I'd like to see
+        this topic addresses in the OEP, even if it just mentions this as a next step.
+
+   * Issue: http://docs.edx.org/ is missing ecommerce docs. Not for the OEP, but where/how do we track doc issues?
+
+* Where should we host a TOC/Index of our docs?
+
+  * Nate got a `draft TOC started here`_.
+
+  * We discussed having a developer-docs repo, and oeps could move here (and we could remove the open-edx-proposals
+    repo).
+
+  * Here is another `Developer Documentation landing page`_ in Confluence.
+
+* Where should documentation live?
+
+   * Possible docs directory per Django App (in addition to per repository).  This was used for oauth_dispatch decisions
+     in edx-platform. This may need some automation/tooling to make this more useful (automated publishing, etc.)
+
+   * OEP vs decision doc.  Just differs in scope?
+
+   * Do we need a README.rst for each Django App? Does it need to live in the root (my preference).  Does it just point
```

Comment:
I would like to see not just each django app but really any important folder in the `edx-platform` codebase (and others) having a `README.rst` file. This is already the case in many places, and it makes browsing the code on GitHub and understanding what's going on much easier.

Examples: https://github.com/edx/edx-platform/tree/master/openedx and https://github.com/edx/edx-platform/tree/master/openedx/features/course_experience

Reply:
Agreed.  I [updated the text](https://github.com/edx/open-edx-proposals/pull/74/files#diff-63e65fcef1f4cd3676f373d1bea15029R153) to indicate that READMEs should be at the top of their corresponding repo/app/folder.